### PR TITLE
feat(android-sdk): handle social error

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* feat(android-sdk): handle social error
* refactor `Completion` interface to receive a generic type for exceptions.
### Note
Implemented according to https://www.notion.so/silverhand/Social-Errors-94f8499b728947fa8309267f69a7d7c5

## Linear Issue Reference

* LOG-2186

## Testing

Test in sample app.

### CODE REVIEW

File: android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/logto/LogtoWebViewSocialHandler.kt [hunk 1]
```diff
@@ -25,7 +27,7 @@ class LogtoWebViewSocialHandler(
             platform: 'android',
             getPostMessage: () => (data) => window.$NAME.postMessage(JSON.stringify(data)),
             supportedSocialConnectorIds: [${getSupportedSocialConnectorIds()}],
-            callbackUriScheme: '${hostActivity.packageName}.logto-callback-web',
+            callbackUriScheme: 'logto-callback://${hostActivity.packageName}/web',
```
Comment (IceHe) [inline android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/logto/LogtoWebViewSocialHandler.kt:30]: should be another PR

Comment (xiaoyijun) [reply android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/logto/LogtoWebViewSocialHandler.kt]: reverted

**Review by IceHe — APPROVED** [review]
LGTM
